packages/broker/test/integration/plugins/operator/contractUtils.ts

```diff
@@ -0,0 +1,206 @@
+        (opts.minimumStakeWei ?? parseEther('60')).toString(),
+        (opts.minHorizonSeconds ?? 0).toString(),
+        (opts.minOperatorCount ?? 1).toString(),
+        opts.streamId,
+        opts.metadata ?? '{}',
+        [
+            chainConfig.contracts.SponsorshipStakeWeightedAllocationPolicy,
+            chainConfig.contracts.SponsorshipDefaultLeavePolicy,
+            chainConfig.contracts.SponsorshipVoteKickPolicy,
+        ], [
+            (opts.earningsPerSecond ?? parseEther('0.01')).toString(),
+            '0',
+            '0'
+        ]
+    )
+    const sponsorshipDeployReceipt = await sponsorshipDeployTx.wait() as ContractReceipt
+    const newSponsorshipEvent = sponsorshipDeployReceipt.events?.find((e) => e.event === 'NewSponsorship')
+    const newSponsorshipAddress = newSponsorshipEvent?.args?.sponsorshipContract
+    const newSponsorship = new Contract(newSponsorshipAddress, sponsorshipABI, opts.deployer) as unknown as Sponsorship
+    return newSponsorship
+}
+
+export function getProvider(): Provider {
+    return new JsonRpcProvider(CHAIN_CONFIG[TEST_CHAIN].rpcEndpoints[0].url)
+}
+
+export function getTokenContract(): TestToken {
+    return new Contract(CHAIN_CONFIG[TEST_CHAIN].contracts.DATA, tokenABI) as unknown as TestToken
+}
+
+interface GenerateWalletWithGasAndTokensOpts {
+    provider?: Provider
+    chainConfig?: { contracts: { DATA: string } }
+    adminKey?: string
+}
+
+export async function generateWalletWithGasAndTokens(opts?: GenerateWalletWithGasAndTokensOpts): Promise<Wallet> {
+    const provider = opts?.provider ?? getProvider()
+    const newWallet = new Wallet(fastPrivateKey())
+    const adminWallet = new Wallet(opts?.adminKey ?? CHAIN_CONFIG[TEST_CHAIN].adminPrivateKey).connect(provider)
+    const token = (opts?.chainConfig !== undefined) 
+        ? new Contract(opts.chainConfig.contracts.DATA!, tokenABI, adminWallet) as unknown as TestToken
+        : getTokenContract().connect(adminWallet)
+    await (await token.mint(newWallet.address, parseEther('1000000'), {
+        nonce: await adminWallet.getTransactionCount()
+    })).wait()
+    await (await adminWallet.sendTransaction({
+        to: newWallet.address,
+        value: parseEther('1')
+    })).wait()
+    return newWallet.connect(provider)
+}
+
+export const delegate = async (delegator: Wallet, operatorContractAddress: string, amount: number, token?: TestToken): Promise<void> => {
+    // onTokenTransfer: the tokens are delegated on behalf of the given data address
+    // eslint-disable-next-line max-len
+    // https://github.com/streamr-dev/network-contracts/blob/01ec980cfe576e25e8c9acc08a57e1e4769f3e10/packages/network-contracts/contracts/OperatorTokenomics/Operator.sol#L233
+    await transferTokens(delegator, operatorContractAddress, amount, delegator.address, token)
+}
+
+export const stake = async (operatorContract: Operator, sponsorshipContractAddresses: string, amount: number): Promise<void> => {
```

Comment:
`sponsorshipContractAddresses` probably should be singular: only one at a time

Reply:
Fixed